files/usr/bin/xlet-about-dialog

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/python3
+
+import sys
+import os
+import json
+import gettext
+import datetime
+import gi
+gi.require_version('Gtk', '3.0')
+from gi.repository import Gtk, GdkPixbuf
+
+usage = """
+usage : xlet-about-dialog applets/desklets/extensions uuid
+"""
+
+gettext.install('cinnamon', '/usr/share/locale')
+
+class AboutDialog(Gtk.AboutDialog):
+    def __init__(self, metadata, stype):
+        super(AboutDialog, self).__init__()
+        self.metadata = metadata
+        self.stype = stype
+
+        self.set_icon()
+
+        if 'version' in metadata:
+            version = self.metadata['version']
+            if 'last-edited' in metadata:
+                timestamp = datetime.datetime.fromtimestamp(self.metadata['last-edited']).isoformat(' ')
+                self.props.version = _("Version %s (%s)") % (version, timestamp)
+            else:
+                self.props.version = _("Version %s") % version
+        elif 'last-edited' in metadata:
+            timestamp = datetime.datetime.fromtimestamp(self.metadata['last-edited']).isoformat(' ')
+            self.props.version = _("Updated on %s") % timestamp
+
+        comments = self._(self.metadata['description'])
+        if 'comments' in self.metadata:
+            comments += '\n' + self.metadata['comments']
+        self.props.comments = comments
+
+        s_id = self.get_spices_id()
+        if s_id:
+            self.props.website = 'https://cinnamon-spices.linuxmint.com/%s/view/%s' % (self.stype, s_id)
+            self.props.website_label = _("More info")
+
+        if 'contributors' in self.metadata:
+            self.props.authors = self.metadata['contributors'].split(',')
+
+        self.props.program_name = '%s (%s)' % (self.metadata['name'], self.metadata['uuid'])
+
+        self.connect('response', self.close)
+        self.show()
+
+    def close(self, *args):
+        Gtk.main_quit()
+
+    def set_icon(self):
+        if 'icon' in self.metadata:
+            self.set_logo_icon_name(self.metadata['icon'])
+        else:
+            icon_path = os.path.join(self.metadata['path'], 'icon.png')
+            if os.path.exists(icon_path):
+                GdkPixbuf.Pixbuf.new_from_file_at_size(icon_path, 128, 128)
```

Comment:
Hello, everybody.

@collinss: Here are two problems:

1. The icon is not set (missing call to `set_logo`). `self.set_logo(GdkPixbuf.Pixbuf.new_from_file_at_size(icon_path, 48, 48))`
2. Icon sizes for xlets (the icon.png file) are limited to 48x48 pixels (I don't remember why). Setting it here to 128x128 pixels results in blurred images ([128x128](https://user-images.githubusercontent.com/3822556/61641946-6dd05900-ac76-11e9-8635-4fda9abe7e45.png) [48x48](https://user-images.githubusercontent.com/3822556/61641948-6dd05900-ac76-11e9-8836-d822800c5f0b.png)).

May I also suggest that you add a call to `self.set_default_icon_name()` at the end of the `set_icon` method? As it is, when the dialog is open, its icon in the panel (the grouped/window list applets) will use the **gtk-missing-icon** (or **image-missing** icon). With the following addition, the icon for the xlet type will be displayed on the panel.

```python
self.set_default_icon_name('cs-%s' % self.metadata['type'])
```

Reply:
Well, I guess I should know better than to submit a pull request at 2am. I remember seeing that I some of the icons were broken, and thought I'd fixed it, but clearly I did not.

Reply:
One issue with setting it the pixel size to 48 is that those icons are much smaller than the ones rendered via icon name, and there doesn't seem to be a way to fix that. It's probably still better than having blurry icons, but I don't know.

Reply:
everything should be working now